From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kleis Auke Wolthuizen <github@kleisauke.nl>
Date: Sat, 12 Nov 2022 14:56:20 +0100
Subject: [PATCH 1/1] Avoid calling gio_win32_appinfo_init during DLL load

Since it scans all XML manifests of UWP apps, which is expensive in terms
of performance. Also, the whole win32appinfo API is unused by libvips or
any of the libraries it relies on.

See: https://gitlab.gnome.org/GNOME/glib/-/issues/2640

Upstream-Status: Inappropriate [disable feature]
Upstream may still depend on win32appinfo.

diff --git a/gio/giomodule.c b/gio/giomodule.c
index 1111111..2222222 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1132,7 +1132,6 @@ DllMain (HINSTANCE hinstDLL,
   if (fdwReason == DLL_PROCESS_ATTACH)
     {
       gio_dll = hinstDLL;
-      gio_win32_appinfo_init (FALSE);
     }
 
   return TRUE;
@@ -1173,7 +1172,6 @@ giomodule_init_ctor (void)
 #ifdef GOBJECT_STATIC_COMPILATION
   gobject_win32_init ();
 #endif /* GOBJECT_STATIC_COMPILATION */
-  gio_win32_appinfo_init (FALSE);
 }
 
 #else /* G_PLATFORM_WIN32 && GIO_STATIC_COMPILATION && !G_HAS_CONSTRUCTORS */
